src/app/store/accounts/blockchain/stacks/stacks-accounts.ts

```diff
@@ -1,23 +1,76 @@
-import { createStacksPublicKey, pubKeyfromPrivKey, publicKeyToAddress } from '@stacks/transactions';
+import { bytesToHex } from '@noble/hashes/utils';
+import { createSelector } from '@reduxjs/toolkit';
+import { HARDENED_OFFSET } from '@scure/bip32';
+import {
+  AddressVersion,
+  createStacksPrivateKey,
+  createStacksPublicKey,
+  getPublicKey,
+  pubKeyfromPrivKey,
+  publicKeyToAddress,
+} from '@stacks/transactions';
+import { deriveStxPrivateKey } from '@stacks/wallet-sdk';
 import { atom } from 'jotai';
 
+import { DATA_DERIVATION_PATH, deriveStacksSalt } from '@shared/crypto/stacks/stacks-address-gen';
+
 import { derivePublicKey } from '@app/common/keychain/keychain';
+import { createNullArrayOfLength } from '@app/common/utils';
 import { storeAtom } from '@app/store';
-import { softwareStacksWalletState } from '@app/store/accounts/blockchain/stacks/stacks-keychain';
+import { selectStacksChain } from '@app/store/chains/stx-chain.selectors';
+import { selectRootKeychain } from '@app/store/in-memory-key/in-memory-key.selectors';
 import { selectLedgerKey } from '@app/store/keys/key.selectors';
 import { addressNetworkVersionState } from '@app/store/transactions/transaction';
 
+import { defaultKeyId } from '../../../keys/key.slice';
 import {
   HardwareStacksAccount,
   SoftwareStacksAccount,
   StacksAccount,
 } from './stacks-account.models';
 
-const softwareAccountsState = atom<Promise<SoftwareStacksAccount[] | undefined>>(async get => {
-  const wallet = await get(softwareStacksWalletState);
+const selectStacksWalletState = createSelector(
+  selectRootKeychain,
+  selectStacksChain,
+  (keychain, chain) => {
+    if (!keychain) return;
+    const { highestAccountIndex, currentAccountIndex } = chain[defaultKeyId];
+    const accountsToRender = Math.max(highestAccountIndex, currentAccountIndex) + 1;
+
```

Comment:
This `map` logic seems a bit tricky and maybe worth extracting to a standalone function?

Reply:
done